### DISCUSSION TITLE
Support for typographic feature tags?

### DISCUSSION THREAD

**Discussion by senneschall** [SixLabors/Fonts#229] [Ideas]
Feature request: support OTF feature tags

For my project, I need to use several OTF feature tags that my font provides: I need to switch between 'ONUM' (oldstyle numbers) and 'LNUM' (lining numbers)
The difference is shown here: https://otf.show/onum and https://otf.show/lnum

Is there a chance this will be added in the future?

**Comment by JimBobSquarePants** [comment]
Interesting...

I do want to add general functionality actually to enable users to explicitly enable feature flags that are not recommended to be enabled by default.  (_[lnum](https://docs.microsoft.com/en-us/typography/opentype/spec/features_ko#tag-lnum) is recommended to be disabled by default so that's why we haven't added it as a default feature_).

To do this we'd have to expose a collection of features (type `Tag`) in `TextOptions` that users can choose to enable by adding entries to. We could then pass those features to the shapers to ensure they are process using the existing infrastructure.

@brianpopow I wonder if we should then exposed your `FeatureTag` enum to make things easier? It's currently unused.

More than happy to accept a PR adding this functionality if you're feeling up to it?

**Reply by brianpopow** [reply]
> To do this we'd have to expose a collection of features (type `Tag`) in `TextOptions` that users can choose to enable by adding entries to. We could then pass those features to the shapers to ensure they are process using the existing infrastructure.
> 

yes, that sounds like a good plan.

> @brianpopow I wonder if we should then exposed your `FeatureTag` enum to make things easier? It's currently unused.

Also yes: using the feature names instead of the acronyms is preferable.
